src/module-config/module-config.ts

```diff
@@ -187,6 +183,18 @@ const setDefaults = (schema, config) => {
       if (!config.hasOwnProperty(key)) {
         config[key] = schema[key]["default"];
       }
+      // We also check if it is an array with object elements, in which case we recurse
+      if (
+        schema[key].hasOwnProperty("arrayElements") &&
+        hasObjectSchema(schema[key].arrayElements)
+      ) {
+        for (let i = 0; i < config[key].length; i++) {
+          config[key][i] = setDefaults(
+            schema[key].arrayElements,
+            config[key][i]
```

Comment:
nit: looks like a good place to use `map` or `forEach`
```suggestion
        const mappedConfig = config[key].map(conf => setDefaults(schema[key].arrayElements, conf);
        config[key] = mappedConfig
```